packages/button/src/ux-button.ts

```diff
@@ -1,124 +1,126 @@
-    this.typeChanged(this.type);
-    this.sizeChanged(this.size);
-    this.effectChanged(this.effect);
-  }
-
-  public typeChanged(newValue: string | null) {
-    const typeClasses = [
-      'ux-button--text',
-      'ux-button--flat',
-      'ux-button--outline',
-      'ux-button--raised',
-      'ux-button--fab'
-    ];
-
-    this.button.classList.remove(...typeClasses);
-
-    if (newValue === 'fab') {
-      this.element.classList.add('ux-fab-button');
-    } else {
-      this.element.classList.remove('ux-fab-button');
-    }
-
-    if (newValue == null || typeClasses.includes(`ux-button--${newValue}`) === false) {
-      newValue = 'raised';
-    }
-
-    this.button.classList.add(`ux-button--${newValue}`);
-  }
-
-  public sizeChanged(newValue: string | null) {
-    const sizeClasses = ['ux-button--small', 'ux-button--medium', 'ux-button--large'];
-
-    this.element.classList.remove(...sizeClasses);
-
-    if (newValue == null || sizeClasses.includes(`ux-button--${newValue}`) === false) {
-      newValue = 'medium';
-    }
-
-    this.element.classList.add(`ux-button--${newValue}`);
-  }
-
-  public effectChanged(newValue: string | null) {
-    const effectClasses = ['ux-button--ripple'];
-
-    this.button.classList.remove(...effectClasses);
-
-    if (newValue == null || effectClasses.includes(`ux-button--${newValue}`) === false) {
-      newValue = 'ripple';
-    }
-
-    this.button.classList.add(`ux-button--${newValue}`);
-  }
-
-  public themeChanged(newValue: any) {
-    if (newValue != null && newValue.themeKey == null) {
-      newValue.themeKey = 'button';
-    }
-
-    this.styleEngine.applyTheme(newValue, this.element);
-  }
-
-  public disabledChanged(newValue: boolean | string) {
-    if (normalizeBooleanAttribute('disabled', newValue)) {
-      this.button.setAttribute('disabled', '');
-    } else {
-      this.button.removeAttribute('disabled');
-    }
-  }
-
-  public onMouseDown(e: MouseEvent) {
-    if (this.button.classList.contains('ux-button--ripple')) {
-      if (this.ripple === null) {
-        this.ripple = new PaperRipple();
-        this.button.appendChild(this.ripple.$);
-      }
-
-      if (this.button.classList.contains('ux-button--fab')) {
-        this.ripple.center = true;
-        this.ripple.round = true;
-      }
-
-      this.ripple.downAction(e);
-    }
-
-    return true;
-  }
-
-  public onMouseUp() {
-    if (this.button.classList.contains('ux-button--ripple') && this.ripple !== null) {
-      this.ripple.upAction();
-    }
-
-    return true;
-  }
-}
+import { customElement, bindable, useView } from 'aurelia-templating';
+import { inject } from 'aurelia-dependency-injection';
+import { StyleEngine, UxComponent, PaperRipple, normalizeBooleanAttribute } from '@aurelia-ux/core';
+import { UxButtonTheme } from './ux-button-theme';
+import { PLATFORM } from 'aurelia-framework';
+
+@inject(Element, StyleEngine)
+@customElement('ux-button')
+@useView(PLATFORM.moduleName('./ux-button.html'))

```

Comment:
why is this needed?

Reply:
I remember we did this, but then had to move back to the original approach with `PLATFORM.moduleName` in index file to ensure compatibility with normal app, and our demo app

Reply:
Don't remember where exactly I saw it, but a team member posted that all plugin developers should explicitly reference templates so that webpack bundles them without the need for `ModuleDependenciesPlugin`

Reply:
bundling an external app, not the demo one, does not pick up templates without this

Reply:
When building the dependency graph of a module, in `index` file, if we do:
```ts
  .globalResources(PLATFORM.moduleName('./ux-button'))
```

Aurelia webpack plugin is taught to start looking for the corresponding html module of `ux-button` module as well, so we don't need to do `PLATFORM.moduleName('ux-button.html')` inside `ux-button` view model module.

If you do:
```ts
import { UxButton } from './ux-button';

// index.ts of ux-button module
  .globalResources(UxButton);
``` 
Then it's a different story. Because there is no `PLATFORM.moduleName` here to tell webpack that when it add `ux-button` as a dependency of `index` module, it should also find corresponding `ux-button.html` and add it as well. Hence the advice.

The above is what I'm guessing based on your comment. It may not be exactly what the other team member said, but it wouldn't be too far off. For that, plus the fact that we do `PLATFORM.moduleName` in our `index` modules, we don't need to double declare the html module of the view model module.

Reply:
Though, there's a case where what you described is beneficial: someone may want to use `UxButton` directly instead of using `.plugin` api:

```ts
import { UxButton } from '@aurelia-ux/button';

export function configure(aurelia) {
  aurelia.use.globalResources(UxButton);
}
```
I think we can try to support than, but let's discourage this usage for now, or at least have better testing story for this first

Reply:
It comes back now... yes it was said in that context. But there is another scenario - multi project repositories with root `node_modules`. By default, the plugin searches for template files in the local `node_modules`. We need to either pass an additional parameter to it, or a plugin needs to explicitly reference the templates :)

Reply:
Ok, that sounds good to me. And given that this change works in all scenarios: demo app, normal app and app nested within a root `node_modules`, I think it's good to go.